File: src/modules/address/domain/dto/address-create.dto.ts
import { IsOptional, IsString } from 'class-validator';

export class AddressCreateDto {
  @IsString({ message: 'Estado inválido!' })
  state: string;

  @IsString({ message: 'Cidade inválida!' })
  city: string;

  @IsString({ message: 'Logradouro inválido!' })
  street: string;

  @IsString({ message: 'Bairro inválido!' })
  neighborhood: string;

  @IsString({ message: 'Complemento inválido!' })
  @IsOptional()
  complement?: string;
}

File: src/main/java/br/com/guizzo/projectbasic/modules/address/domain/dto/AddressCreateDTO.java
package br.com.guizzo.projectbasic.modules.address.domain.dto;

import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class AddressCreateDTO{
        @NotNull(message = "Estado inválido!")
        private final String state;

        @NotNull(message = "Cidade inválida!")
        private final String city;

        @NotNull(message = "Logradouro inválido!")
        private final String street;

        @NotNull(message = "Bairro inválido!")
        private final String neighborhood;

        private final String complement;
}